packages/bot-web-ui/src/stores/run-panel-store.js

```diff
@@ -215,18 +221,28 @@ export default class RunPanelStore {
     @action.bound
     onBotRunningEvent() {
         this.is_stop_button_disabled = false;
+        const { self_exclusion } = this.root_store;
         this.has_open_contract = true;
+
+        if (self_exclusion.run_limit !== -1) {
```

Comment:
Could we do a check for `is_restricted` instead?

```JS
if (is_restricted) {
    self_exclusion.run_limit -= 1;

    if (self_exclusion.run_limit < 0) {
        this.onStopButtonClick();
    }
}
```

Reply:
is restricted also use to show and hide modal because of that I've update code and now we have a new computed to check should bot run in self-exclusion-store 
@computed 
get should_bot_run